### DISCUSSION TITLE
Queue Changes

### DISCUSSION THREAD

**Discussion by TUBEOF** [spotify-web-api-java/spotify-web-api-java#335] [General]
Hi, i just updated to new v8 and saw this PR
https://github.com/spotify-web-api-java/spotify-web-api-java/pull/329

Before that, it was possible to get more information, like the artists using `track.getArtists().getName());` for example.
Did I miss something, or is it now required to fetch this information using a GetTrack call using the track id?

**Comment by TUBEOF** [comment]
@dargmuesli Just once a friendly reminder as I see that was replied to in another thread, just to make sure this one wasn't overlooked :D

**Reply by dargmuesli** [reply]
Hey, I reply to thread for which I know what I could contribute. While I maintain this repo for free, I don't feel required to answer every question. I feel more responsible when it comes to bug reports, but that's another story. Coming back to your question: for one reading your question it's not clear what `track` you are speaking about. It seems you haven't gotten it by running `GetTrack` as you ask if that would be required now.

For people to be able to help you they need a clear example of what does not work. In your question it's only implicitly understandable if one knew which `track` you write about. While it is more work for you, please always provide a code example or even a repository showing the exact problem so you do everyone who tries to help you a favor by removing the burden of figuring out the code example for you from them :raised_hands:

**Comment by Selbi182** [comment]
I'm a bit late, but I hope I can still help since I submitted the original PR. The old way assumed that the queue only contains tracks, which is not the case; you can also queue up podcasts. So, I changed it to IPlaylistItem, this Java wrapper's version of abstractly talking about a track or podcast.

The whole secret is to simply check the ModelObjectType and then cast it to the appropriate class. For your original example, it could be done like this:

```
PlaybackQueue playbackQueue = spotifyApi.getTheUsersQueue().build().execute();
List<IPlaylistItem> queueItems = playbackQueue.getQueue();
IPlaylistItem first = queueItems.get(0);
if (ModelObjectType.TRACK.equals(first.getType())) {
  ArtistSimplified[] artists = ((Track) first).getArtists();
  System.out.println(artists[0].getName());
}
```

Yes, it isn't as pretty as before, but you can thank Spotify for being so confusing about mixing tracks with podcasts in the first place.
